src/libstore/build.cc

```diff
@@ -4248,6 +4286,7 @@ void DerivationGoal::done(BuildResult::Status status, std::optional<Error> ex)
     mcRunningBuilds.reset();
 
     if (result.success()) {
+        registerOutputMappings();
```

Comment:
There might be a better place for this indeed.
I didn't put it in `registerOutputs` because we want it to be called even when no build took place (either because we substituted the output or because we've short-cutted the build after resolving the derivation).